libs/books/feature/src/lib/book-search/book-search.component.spec.ts

```diff
@@ -1,27 +1,63 @@
-import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { async, ComponentFixture, TestBed, tick, fakeAsync, discardPeriodicTasks } from '@angular/core/testing';
 import { NoopAnimationsModule } from '@angular/platform-browser/animations';
-import { SharedTestingModule } from '@tmo/shared/testing';
-
+import { createBook, SharedTestingModule } from '@tmo/shared/testing';
 import { BooksFeatureModule } from '../books-feature.module';
 import { BookSearchComponent } from './book-search.component';
+import { MockStore, provideMockStore } from '@ngrx/store/testing'
+import { 
+  addToReadingList,
+  searchBooks,
+  getAllBooks,
+  getBooksError,
+  getBooksLoaded,
+} from '@tmo/books/data-access'
+import { By } from '@angular/platform-browser';
+import { dispatch } from 'rxjs/internal/observable/pairs';
 
-describe('ProductsListComponent', () => {
+describe('BookSearchComponent', () => {
   let component: BookSearchComponent;
   let fixture: ComponentFixture<BookSearchComponent>;
+  let dispatchSpy;
+  let store: MockStore
 
   beforeEach(async(() => {
     TestBed.configureTestingModule({
-      imports: [BooksFeatureModule, NoopAnimationsModule, SharedTestingModule]
+      imports: [BooksFeatureModule, NoopAnimationsModule, SharedTestingModule],
+      providers: [provideMockStore()]
     }).compileComponents();
   }));
 
   beforeEach(() => {
     fixture = TestBed.createComponent(BookSearchComponent);
     component = fixture.componentInstance;
+    store = TestBed.inject(MockStore);
+    dispatchSpy = jest.spyOn(store, 'dispatch');
+    store.overrideSelector(getAllBooks, [
+      {...createBook('A'), isAdded: false},
+      {...createBook('AB'), isAdded: false}
+    ]);
     fixture.detectChanges();
   });
 
   it('should create', () => {
     expect(component).toBeDefined();
   });
+
+  it('should call dispatch action searchBooks after 500ms when the term is changed and not before', fakeAsync(() => {
```

Comment:
Provide new line in test cases after assignment, action and assertion.

Keep only one action per test case and cover both positive and negative scenario.